### ISSUE TITLE
Navi3x test_verify_gemm failure: LLVM ERROR: Cannot select: t152: v2f16,ch = BUFFER_ATOMIC_FADD

### ISSUE THREAD

**Issue by ahsan-ca** [ROCm/AMDMIGraphX#3198]
`gemm_add<migraphx::shape::half_type>` test fails with the following error:

```
LLVM ERROR: Cannot select: t152: v2f16,ch = BUFFER_ATOMIC_FADD<(volatile dereferenceable load store (s32) on %ir.220, align 1, !alias.scope !18, !noalias !19, addrspace 8)> # D:1 t115, t158, t146, Constant:i32<0>, t114, Constant:i32<0>, TargetConstant:i32<0>, TargetConstant:i32<0>, TargetConstant:i1<0>
  t158: v2f16 = bitcast # D:1 t157
    t157: i32 = select # D:1 t124, t155, t156
      t124: i1 = truncate # D:1 t93
        t93: i32 = select # D:1 t89, t86, Constant:i32<640>
          t89: i1 = setcc # D:1 t77, Constant:i32<2>, setult:ch
            t77: i32 = or disjoint # D:1 t76, t72
              t76: i32 = AssertZext # D:1 t74, ValueType:ch:i7
                t74: i32,ch = CopyFromReg # D:1 t0, Register:i32 %11
                  t73: i32 = Register %11
              t72: i32 = and # D:1 t6, Constant:i32<1>
                t6: i32 = srl # D:1 t4, Constant:i32<4>
                  t4: i32 = AssertZext # D:1 t2, ValueType:ch:i8
```

**Comment by ahsan-ca** [comment]
Reproduced the error locally.

Did the following:

1. Tried building the test with latest MIGraphX (2b4d328baea4ecaa47956ec9562f9fec79f82acb), and used latest rocMLIR changes (ec52b22fa2b68277803190979d39ae7aa1e873d5) and the test case passed successfully.

2. Tried building the test with latest MIGraphX (2b4d328baea4ecaa47956ec9562f9fec79f82acb), and used rocMLIR commit that MIGraphX currently uses (5261f9fb8cc5e0109cf59bbf4cb11cc6fb527209) and the test case passed successfully.

3. Tried mainline build `14236` (`6.2` http://rocm-ci.amd.com/job/compute-rocm-dkms-no-npi-hipclang/14236/) and test case fails.

4. Now trying mainline build `14273` (`6.3` http://rocm-ci.amd.com/job/compute-rocm-dkms-no-npi-hipclang/14273/).

**Comment by ahsan-ca** [comment]
Passes with mainline build: http://rocm-ci.amd.com/job/compute-rocm-dkms-no-npi-hipclang/14273/ 

./bin/test_verify "test_gemm_add<migraphx::shape::half_type>"
[   RUN    ] test_gemm_add<migraphx::shape::half_type>
Skipped 86 configs for gpu::mlir_op
[ COMPLETE ] test_gemm_add<migraphx::shape::half_type> (20644.6ms)
[==========] 1 tests ran

**Comment by ahsan-ca** [comment]
This passes in 6.3. We need to backport the following commit to make it 6.2 pass:

commit a3593f6cd50bab4448c74860bec8312032f15569 (HEAD)
Author: Umang Yadav <29876643+umangyadav@users.noreply.github.com>
Date:   Sat Jun 8 09:39:43 2024 -0400

    update rocMLIR SHA to pull in bug fix for invalid block size on Navi3x (#3159)
